File: leetcode/additive-number.js
function isAdditiveNumber(s) {
    function isValid(num1, num2, remaining) {
        if (remaining.length === 0) {
            return true;
        }

        const sum = (BigInt(num1) + BigInt(num2)).toString();
        if (remaining.startsWith(sum)) {
            return isValid(num2, sum, remaining.slice(sum.length));
        } else {
            return false;
        }
    }

    const length = s.length;
    for (let i = 1; i < length / 2 + 1; i++) {
        for (let j = i + 1; j < length; j++) {
            const num1 = s.slice(0, i);
            const num2 = s.slice(i, j);

            if ((num1.length > 1 && num1[0] === '0') || (num2.length > 1 && num2[0] === '0')) {
                continue;  // Invalid leading zeros
            }

            if (isValid(num1, num2, s.slice(j))) {
                return true;
            }
        }
    }

    return false;
}

// Example usage:
const s1 = "112358";
const s2 = "199100199";
console.log(isAdditiveNumber(s1));  // Output: true
console.log(isAdditiveNumber(s2));  // Output: true

File: leetcode/additive-number.py
class Solution:
    def isAdditiveNumber(self, num: str) -> bool:
        def is_valid(num1, num2, remaining):
            if not remaining:
                return True

            sum_str = str(int(num1) + int(num2))
            if remaining.startswith(sum_str):
                return is_valid(num2, sum_str, remaining[len(sum_str):])
            else:
                return False

        length = len(num)
        for i in range(1, length // 2 + 1):
            for j in range(i + 1, length):
                num1, num2 = num[:i], num[i:j]
                if (len(num1) > 1 and num1[0] == '0') or (len(num2) > 1 and num2[0] == '0'):
                    continue  # Invalid leading zeros

                if is_valid(num1, num2, num[j:]):
                    return True

        return False

# Example usage:
sol = Solution()
print(sol.isAdditiveNumber("112358"))  # Output: True
print(sol.isAdditiveNumber("199100199"))  # Output: True